mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -139,17 +141,56 @@ def _get_run_artifact_root(self, run_id):
         run_response = self._call_endpoint(MlflowService, GetRun, json_body)
         return run_response.run.info.artifact_uri
 
-    def _get_write_credentials(self, run_id, path=None):
-        json_body = message_to_json(GetCredentialsForWrite(run_id=run_id, path=path))
-        return self._call_endpoint(
-            DatabricksMlflowArtifactsService, GetCredentialsForWrite, json_body
-        )
+    def _get_credential_infos(self, request_message_class, run_id, paths):
+        """
+        Issue one or more requests for artifact credentials, providing read or write
+        access to the specified run-relative artifact `paths` within the MLflow Run specified
+        by `run_id`. The type of access credentials, read or write, is specified by
+        `request_message_class`.
 
-    def _get_read_credentials(self, run_id, path=None):
-        json_body = message_to_json(GetCredentialsForRead(run_id=run_id, path=path))
-        return self._call_endpoint(
-            DatabricksMlflowArtifactsService, GetCredentialsForRead, json_body
-        )
+        :return: A list of `ArtifactCredentialInfo` objects providing read access to the specified
+                 run-relative artifact `paths` within the MLflow Run specified by `run_id`.
+        """
+        credential_infos = []
+        page_token = None
+
+        for paths_chunk in chunk_list(paths, _MAX_CREDENTIALS_REQUEST_SIZE):
+            while True:
+                if page_token:
+                    json_body = message_to_json(
+                        request_message_class(
+                            run_id=run_id, path=paths_chunk, page_token=page_token
```

Comment:
If the page_token defaults to "" instead of None, could the if-else clause in this block be removed?

Reply:
Unfortunately, it doesn't seem to be the case. Specifying an empty string results in:

```
mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Malformed token supplied with artifact credentials request.
```

However, `None` seems to work! Not sure why I had this logic here previously. I've removed the if / else here and in the definition of `list_artifacts()`.